common/txmgr/types/keystore.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+import "github.com/smartcontractkit/chainlink/core/services/pg"
+
+// KeyStore encompasses the subset of keystore used by txmgr
+type KeyStore[ADDR any, ID any, TX any, META any] interface {
+	CheckEnabled(address ADDR, chainID ID) error
+	GetNextMetadata(address ADDR, chainID ID, qopts ...pg.QOpt) (META, error)
+	GetEnabledAddressesForChain(chainId ID) ([]ADDR, error)
```

Comment:
Maybe drop the `Get*` prefixes?
```suggestion
	NextMetadata(address ADDR, chainID ID, qopts ...pg.QOpt) (META, error)
	EnabledAddressesForChain(chainId ID) ([]ADDR, error)
```